dexcom/event.go

```diff
@@ -196,44 +217,69 @@ func (e *Event) Validate(validator structure.Validator) {
 			e.validateHealth(validator)
 		case EventTypeInsulin:
 			e.validateInsulin(validator)
+		case EventTypeNotes:
+			e.validateNote(validator)
+		case EventTypeBG:
+			e.validateBG(validator)
+		case EventTypeUnknown:
+			e.validateUnkown(validator)
 		}
 	}
-	validator.String("eventId", e.ID).Exists().NotEmpty()
+	validator.String("recordId", e.ID).Exists().NotEmpty()
 	validator.String("eventStatus", e.Status).Exists().OneOf(EventStatuses()...)
+	validator.String("transmitterId", e.TransmitterID).Exists().Using(TransmitterIDValidator)
+	validator.String("transmitterGeneration", e.TransmitterGeneration).Exists().OneOf(DeviceTransmitterGenerations()...)
+	validator.String("displayDevice", e.DisplayDevice).Exists().OneOf(DeviceDisplayDevices()...)
 }
 
 func (e *Event) validateCarbs(validator structure.Validator) {
 	validator.String("eventSubType", e.SubType).NotExists()
-	if e.Unit != nil || e.Value != nil {
-		validator.String("unit", e.Unit).Exists().OneOf(EventUnitCarbsGrams)
-		validator.Float64("value", e.Value).Exists().InRange(EventValueCarbsGramsMinimum, EventValueCarbsGramsMaximum)
+	validator.String("unit", e.Unit).Exists().OneOf(EventUnitCarbsGrams)
+	validator.String("value", e.Value).Exists().NotEmpty()
+	if e.Value != nil {
+		floatVal, _ := strconv.ParseFloat(*e.Value, 64)
+		validator.Float64("value", &floatVal).Exists().InRange(EventValueCarbsGramsMinimum, EventValueCarbsGramsMaximum)
 	}
 }
 
 func (e *Event) validateExercise(validator structure.Validator) {
-	// HACK: Dexcom - value of -1 is invalid; ignore unit and value instead (per Dexcom)
-	if e.Value != nil && *e.Value == -1.0 {
```

Comment:
I optimistically removed this ... we are now dealing with strings but again might have to do some focused testing for this